packages/webapp/pages/_app.tsx

```diff
@@ -95,14 +97,20 @@ function InternalApp({ Component, pageProps, router }: AppProps): ReactElement {
       tokenRefreshed &&
       user &&
       !user.infoConfirmed &&
-      window.location.pathname.indexOf('/register') !== 0
+      window.location.pathname.indexOf(REGISTRATION_PATH) !== 0
     ) {
+      const referrer = document.referrer && new URL(document.referrer);
+      if (referrer.pathname?.indexOf(REGISTRATION_PATH) !== -1) {
+        window.location.reload();
+        return;
+      }
```

Comment:
Please add a comment what exactly this check does and why

Reply:
Yes, I have tested locally multiple times. If it came from the registration and the local state/data has not been cleared, we will have to force a `reload`. I have also tried all different combinations including on extensions.

This should not produce a loop since the `document.referrer` will change once the the reload has happened, so it won't be valid since it is not the registration path anymore. 
Coming from the registration page? The condition above this checks if it should not enter this block by validating if it is currently the registration page we are looking at.

The only loop-like behavior will be if the user doesn't fill-up the form (`user.infoConfirmed = false`) and keeps on clicking our dailydev logo (wants to go to the home page) which is our expected behavior since we want to finish up the form first before proceeding with the app if you are trying to register.